### ISSUE TITLE
Non-deterministic failure of machi_merkle_tree_test

### ISSUE THREAD

**Issue by slfritchie** [basho/machi#52]
On `master` branch at commit e9b1134cd970af79d5d3c51e2f25b02273d5f1ac

```
% sh -c 'for i in `seq 1 10`; do echo number $i; rebar skip_deps=true eunit suites=machi_merkle_tree_test; done'
number 1
==> machi (eunit)
  Test passed.
number 2
==> machi (eunit)
  Test passed.
number 3
==> machi (eunit)
  Test passed.
number 4
==> machi (eunit)
  Test passed.
number 5
==> machi (eunit)
machi_merkle_tree_test: basic_test (module 'machi_merkle_tree_test')...*failed*
in function machi_merkle_tree_test:'-basic_test/0-fun-1-'/3 (test/machi_merkle_tree_test.erl, line 45)
**error:{assertEqual_failed,[{module,machi_merkle_tree_test},
                     {line,45},
                     {expression,"length ( machi_merkle_tree : naive_diff ( T1 , T2 ) )"},
                     {expected,1},
                     {value,8}]}
```

**Comment by slfritchie** [comment]
Test comment (please ignore)

**Comment by jadeallenx** [comment]
Not sure how I didn't see this already, but yes, it's a legitimate problem. I think the proposed fix in #53 is fine. Thanks.

**Comment by slfritchie** [comment]
Fixed by #53, closing.
